### ISSUE TITLE
tf.keras.layers.SimpleRNN train with batch data  like original keras API

### ISSUE THREAD

**Issue by sunzhe09** [tensorflow/tensorflow#33908]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):conda
- TensorFlow version (use command below):1.14.0
- Python version:3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:10.1
- GPU model and memory:16G

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
I can not train my image with shape (None,256,256)

**Describe the expected behavior**
I need a interface from the new api like this:
# RNN cell
model.add(SimpleRNN(
    # for batch_input_shape, if using tensorflow as the backend, we have to put None for the batch_size.
    # Otherwise, model.evaluate() will get error.
    batch_input_shape=(None, TIME_STEPS, INPUT_SIZE),  # Or: input_dim=INPUT_SIZE, input_length=TIME_STEPS,
    output_dim=CELL_SIZE,
    unroll=True,
))

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.
class Simple_IRNN(tf.keras.layers.Layer):
    def __init__(self,in_channels,scope='simpleirnn',alpha=1.0):
        super(Simple_IRNN,self).__init__()
        self.num_outputs=in_channels
        self.batch_size=8
        self.num_outputs=256
        self.alpha=alpha
    

        self.left=tf.keras.layers.SimpleRNN(self.num_outputs,activation='relu',
        recurrent_initializer=tf.keras.initializers.Identity(gain=self.alpha),return_sequences=True) # (batch_size, timesteps,input_features)
        self.right=tf.keras.layers.SimpleRNN(self.num_outputs,activation='relu',
        recurrent_initializer=tf.keras.initializers.Identity(gain=self.alpha),return_sequences=True)
        self.up=tf.keras.layers.SimpleRNN(self.num_outputs,activation='relu',
        recurrent_initializer=tf.keras.initializers.Identity(gain=self.alpha),return_sequences=True)
        self.down=tf.keras.layers.SimpleRNN(self.num_outputs,activation='relu',
        recurrent_initializer=tf.keras.initializers.Identity(gain=self.alpha),return_sequences=True)

    def call(self,inputs):
   

        inputs=tf.nn.relu(inputs)
        inputs=tf.reduce_sum(inputs,axis=-1)
  
        mask=tf.cast(tf.ones(shape=(256,256),dtype=tf.int32),dtype=tf.bool)

        temp=tf.image.flip_up_down(inputs)
        temp=tf.image.rot90(temp,k=1)

        print("inputs shape:",inputs.shape)
        self.up.get_initial_state(inputs=temp)   
        self.right.get_initial_state(inputs=inputs)

        output_up=self.up(temp)
        output_right=self.right(inputs)

        temp=tf.image.rot90(inputs,k=1)

        self.down.get_initial_state(inputs=temp)
        output_down=self.down(temp)
       

        temp=tf.image.flip_left_right(inputs)
        self.left.get_initial_state(inputs=temp)
  
        output_left=self.left(temp)

        output_right=tf.reshape(output_right,(self.batch_size,256,256,1))
        output_up=tf.reshape(output_up,(self.batch_size,256,256,1))
        output_down=tf.reshape(output_down,(self.batch_size,256,256,1))
        output_left=tf.reshape(output_left,(self.batch_size,256,256,1))

        
        return output_up,output_right,output_down,output_left

here is my selfdefine layer ,it can run,but I found change to my selflayer my eval loss and acc don't change anymore 

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by jvishnuvardhan** [comment]
@sunzhe09 Can you please provide a standalone code to reproduce the issue? Thanks!

**Comment by sunzhe09** [comment]
@jvishnuvardhan I can run my code ,because my batchsize is None in the Old code,but when I train ,my loss didn't change

**Comment by jvishnuvardhan** [comment]
@sunzhe09 Can you please provide a standalone code to reproduce the issue? Thanks!

**Comment by sunzhe09** [comment]
@jvishnuvardhan sorry for my late response,I have pass the code ,it't my fault.when I set a new model with Input layer and set the input shape then, the code can run .The rnn return sequence can't process tensor with None batch
